src/server/pachyderm_test.go

```diff
@@ -4656,6 +4656,48 @@ func TestSelfReferentialPipeline(t *testing.T) {
 	))
 }
 
+func TestPipelineBadImage(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration tests in short mode")
+	}
+	t.Parallel()
+	c := getPachClient(t)
+	pipeline1 := uniqueString("bad_pipeline")
+	require.NoError(t, c.CreatePipeline(
+		pipeline1,
+		"BadImage",
+		[]string{"true"},
+		nil,
+		nil,
+		client.NewCronInput("time", "@every 20s"),
+		"",
+		false,
+	))
+	pipeline2 := uniqueString("bad_pipeline")
+	require.NoError(t, c.CreatePipeline(
+		pipeline2,
+		"bs/badimage:vcrap",
+		[]string{"true"},
+		nil,
+		nil,
+		client.NewCronInput("time", "@every 20s"),
+		"",
+		false,
+	))
+	require.NoError(t, backoff.Retry(func() error {
+		for _, pipeline := range []string{pipeline1, pipeline2} {
+			pipelineInfo, err := c.InspectPipeline(pipeline)
+			if err != nil {
+				return err
+			}
+			if pipelineInfo.State != pps.PipelineState_PIPELINE_FAILURE {
```

Comment:
Shouldn't we test for the `reason` too?

Reply:
Seems like a reasonable idea, I think I'm just going to test for it being non empty though rather than a particular error message since the message might change.

Reply:
Well if the message changes we can just update the test.  See how I tested for job `reason`.  I basically check for a substring.

Reply:
So I tried checking for a substring, it passed for me locally but then on CI the substring couldn't be found. I'm unclear how but somehow even though it's the same version of kubernetes running in both environment they seem to return different error strings. It doesn't really seem worth it to me to go down a CI debugging rabbit hole just to check this string. It's not really clear to me how much more coverage we'll be getting out of our tests if they check some substring for the Reason.